common/date_format.py

```diff
@@ -15,80 +16,85 @@
-#Nonstring
-NONSTRING = -1
 
-#Has leading zero
+#Has a two-digit number with a leading zero (e.g., 02, 08)
 leading_zero = re.compile("(^|\s+)0[0-9]\s+")
-#Does not have leading zero
+#Has a single-digit number that has no leading zero (e.g., 3, 7)
 no_leading_zero = re.compile("(^|\s+)[1-9]\s+")
 
-class DateFormat:
-    def __init__(self, date_string):
-        self.date_string = date_string
-        # Binary representation of a set of atts
-        self.date_hash = self.att_binary(date_string)
-        # 1 represents 0 seen, 2 represents no zero
-        self.zeros_seen = set()
-        # Add zero status to seen
-        zero_stat = self.zero_status(date_string)
-        self.add_zero_status(zero_stat)
-
-    def same_hash(self, other):
-        if (self.date_hash != other.date_hash):
-            return False
-        return True
-
-    def add_zero_status(self, status):
-        self.zeros_seen.add(status)
-
-    @staticmethod
-    def att_binary(date_string):
-        if not isinstance(date_string, str):
-            return NONSTRING
-        att_binary = 0
-        for att in [att1, att2, att3, att4, att5, att6, att7]:
-            if (re.search(att, date_string) is not None):
-                att_binary = att_binary | 1
-            att_binary = att_binary << 1
-        return att_binary >> 1
-
-    @staticmethod
-    def zero_status(date_string):
-        if not isinstance(date_string, str):
-            return 0
-        if (re.search(leading_zero, date_string) is not None):
-            return 1
-        elif (re.search(no_leading_zero, date_string) is not None):
-            return 2
-        return 0
-
-class DateData:
-    def __init__(self, dates=[]):
-        self.dates = dates
-        self.seen_formats = {}
-
-    def process_date(self, date):
-        self.dates.append(date)
-        curr_format = DateFormat(date)
-
-        is_new_format = True
-        curr_zero = list(curr_format.zeros_seen)[0]
-        for seen_format in self.seen_formats:
-            if (curr_format.same_hash(seen_format)):
-                is_new_format = False
-                self.seen_formats[seen_format] += 1
-                seen_format.add_zero_status(curr_zero)
-                break
-        if (is_new_format):
-            self.seen_formats[curr_format] = 1
+def date_to_template(date_string):
+    """Returns a binary templates for a date string."""
+    binary_template = 0
+    for att in [att1, att2, att3, att4, att5, att6, att7]:
+        if re.search(att, date_string):
+            binary_template = binary_template | 1
+        binary_template = binary_template << 1
+    binary_template = binary_template >> 1
+    return binary_template
+
+def has_leading_zero(date_string):
+    return re.search(leading_zero, date_string)
+
+def has_no_leading_zero(date_string):
+    return re.search(no_leading_zero, date_string)
+
+def log_if_inconsistent(date_string):
+    if has_leading_zero(date_string) and has_no_leading_zero(date_string):
```

Comment:
Yeah, just a safeguard.  I *think* it should be impossible for dates.  The case where it would trigger would be something like `blah 07 6 blah`.  But a date should only contain a single number that is could possibly be single digit, so it fels impossible.

I'll add a comment to the code.